recipes/openscenegraph/all/conanfile.py

```diff
@@ -0,0 +1,538 @@
+            setup_plugin("png").requires.extend(("libpng::libpng", "zlib::zlib"))
+
+        if self.options.with_tiff:
+            setup_plugin("tiff").requires.append("libtiff::libtiff")
+
+        if self.options.with_gdal:
+            setup_plugin("gdal").requires.extend(("osgTerrain", "gdal::gdal"))
+            setup_plugin("ogr").requires.append("gdal::gdal")
+
+        if self.options.with_gta:
+            setup_plugin("gta").requires.append("libgta::libgta")
+
+        # 3D Image plugins
+        if self.options.get_safe("with_dcmtk"):
+            plugin = setup_plugin("dicom")
+            plugin.requires.extend(("osgVolume", "dcmtk::dcmtk"))
+            if self.settings.os == "Windows":
+                plugin.system_libs = ["wsock32", "ws2_32"]
+
+        # 3rd party 3d plugins
+        setup_plugin("3dc")
+        setup_plugin("p3d").requires.extend(("osgGA", "osgText", "osgVolume", "osgFX", "osgViewer", "osgPresentation"))
+
+        if self.options.with_curl:
+            plugin = setup_plugin("curl")
+            plugin.requires.append("libcurl::libcurl")
+            if self.options.with_zlib:
+                plugin.requires.append("zlib::zlib")
+
+        if self.options.with_zlib:
+            setup_plugin("gz").requires.append("zlib::zlib")
+
+        # with_inventor
+        # setup_plugin("iv")
+
+        # with_collada
+        # setup_plugin("dae")
+
+        # with_fbx
+        # setup_plugin("fbx")
+
+        # with_opencascade
+        # setup_plugin("opencascade")
+
+        setup_plugin("bvh").requires.append("osgAnimation")
+        setup_plugin("x")
+        setup_plugin("dxf").requires.append("osgText")
+        setup_plugin("openflight").requires.append("osgSim")
+        setup_plugin("obj")
+        setup_plugin("pic")
+        setup_plugin("stl")
+        setup_plugin("3ds")
+        setup_plugin("ac")
+        setup_plugin("pov")
+        setup_plugin("logo")
+        setup_plugin("lws")
+        setup_plugin("md2")
+        setup_plugin("osgtgz")
+        setup_plugin("tgz")
+        setup_plugin("shp").requires.extend(("osgSim", "osgTerrain"))
+        setup_plugin("txf").requires.append("osgText")
+        setup_plugin("bsp")
+        setup_plugin("mdl")
+        setup_plugin("gles").requires.extend(("osgUtil", "osgAnimation"))
+        setup_plugin("osgjs").requires.extend(("osgAnimation", "osgSim"))
+        setup_plugin("lwo").requires.append("osgFX")
+        setup_plugin("ply")
+        setup_plugin("txp").requires.extend(("osgSim", "osgText"))
+
+        # with_ffmpeg
+        # setup_plugin("ffmpeg")
+
+        # with_gstreamer
+        # setup_plugin("gstreamer")
+
+        # with_directshow
+        # setup_plugin("directshow")
+
+        if tools.is_apple_os(self.settings.os):
+            setup_plugin("imageio").frameworks = ["Accelerate"]
+
+        if ((self.settings.os == "Macos" and self.settings.os.version and tools.Version(self.settings.os.version.value) >= "10.8")
```

Comment:
So I'm pretty sure originally I didn't include the check for the presence of `os.version`, which caused issues with the CI or with the default Macos profile as at least one of those didn't include the version, so `tools.Version` would complain about `None` values.

I also had tried not using the `.value` on the version and found that it broke with similar issues.

Reply:
Would that not mean in CCI it's always false and the extra plugin will never be added?

Reply:
Correct, but that is the price that is payed for ignoring the version, otherwise a user might get binaries that just don't work on their system. This is better than the status quo on linux, where a user can easily shoot themselves in the foot with CCI binaries due to glibc versions (our team bans all CCI linux binaries for this very reason).

A user who wants to use the plugin will just need a profile that includes the Macos version.

Reply:
> glibc versions

Hopefully the tribe can change that! 

Okay, we'll see what the CCI team says but that would take a more advanced consumer 🤔 I dont have an issue. I think this is a very good starting point.